src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -103,6 +103,15 @@ private DateTimeOffset GetOffsetOrDefault(IDictionary props, string field, DateT
         throw ex;
     }
 
+    private TimeSpan GetDurationOrDefault(IDictionary props, TimeSpan defaultValue = default)
```

Comment:
Should we add this as an extension method on IDictionary?  something like:
`T GetValueOrDefault<T>(this IDictionary dictionary, string key, T defaultValue = default)`